### PR TITLE
[SandboxIR][Bench] SandboxIR creation

### PR DESCRIPTION
**Pull request by vporpo**
Adds a benchmark for the overhead of SandboxIR creation.

### CODE REVIEW

**Comment by alinas** [comment]
Can you rebase on top of https://github.com/llvm/llvm-project/pull/108273?

**Comment by vporpo** [comment]
> Can you rebase on top of https://github.com/llvm/llvm-project/pull/108273?

It is rebased as far as I can tell.

**Comment by alinas** [comment]
Can you update the names throughout the patch following that change?
s/SBox/SBoxNoTracking
s/SBoxTr/SBoxTracking

**Comment by vporpo** [comment]
Oops let me fix them.

File: llvm/benchmarks/SandboxIRBench.cpp [hunk 1]
```diff
@@ -89,6 +89,31 @@ static std::string generateBBWalkIR(unsigned Size) {
   return SS.str();
 }
 
+template <IR IRTy> static void SBoxIRCreation(benchmark::State &State) {
+  static_assert(IRTy != IR::LLVM, "Expected SBox and SBoxTr");
```
Comment (alinas) [inline llvm/benchmarks/SandboxIRBench.cpp:93]: Update the assert message.

**Review by alinas — APPROVED** [review]

**Pull request merged**